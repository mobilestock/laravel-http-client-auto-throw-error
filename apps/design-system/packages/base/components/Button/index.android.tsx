/** @format */

import React, { PropsWithChildren } from 'react'
import { ActivityIndicator, StyleProp, TextStyle, TouchableHighlightProps } from 'react-native'
import styled, { css } from 'styled-components/native'

import { globalTema } from '../../utils/theme-native'

export interface PropsButton extends TouchableHighlightProps {
  text?: string
  isLoading?: boolean
  textStyle?: StyleProp<TextStyle>
}

export const Button: React.FC<PropsWithChildren<PropsButton>> = props => {
  return (
    <ButtonStyle {...props}>
      {props.isLoading ? (
        <ActivityIndicator color={globalTema.cores.branco} size={25} />
      ) : (
        <>{props.children ? props.children : <Text style={props.textStyle}>{props.text}</Text>}</>
      )}
    </ButtonStyle>
  )
}
const ButtonStyle = styled.TouchableHighlight`
  background-color: ${({ theme }) => theme.cores.corSecundaria};
  min-height: ${({ theme }) => theme.layout.height(2)}px;
  justify-content: center;
  align-items: center;
  border-radius: ${({ theme }) => theme.layout.size(2)}px;
  margin: ${({ theme }) => theme.layout.size(1)}px;

  ${({ disabled }) =>
    disabled &&
    css`
      opacity: 0.5;
    `}
`

const Text = styled.Text`
  color: ${({ theme }) => theme.cores.branco};
  font-size: ${({ theme }) => theme.fonts.size(16)}px;
  text-align: center;
`
