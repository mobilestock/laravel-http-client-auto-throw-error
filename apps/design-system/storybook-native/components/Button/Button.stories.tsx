import { Button } from "@mobilestock/base/index.android";
import type { Meta, StoryObj } from "@storybook/react";
import React from "react";
import { View } from "react-native";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";

Button.defaultProps = {
    isLoading: false,
    text: 'Clique aqui',
    style: {
      width: 200,
      height: 50,
      backgroundColor: theme.cores.azul40
    }
  }

const meta = {
  title: "Botão",
    component: Button,
    parameters: {
      notes: `
        Exemplo de código:

        <Button
            isLoading={false}
            text="Clique aqui"
        />
      `,
  },
  args: {
    text: "Clique aqui",
  },
  decorators: [
    (Story) => (
      <View style={{ padding: 16 }}>
        <ThemeProvider theme={theme}>
          <Story />
        </ThemeProvider>
      </View>
    ),
    ],
    argTypes: {
        isLoading: {
          control: 'boolean',
          description: 'Indica se o botão está em estado de carregamento',
          defaultValue: false
        },
        text: {
          control: 'text',
          description: 'Texto exibido no botão',
          defaultValue: 'Clique aqui'
        },
        disabled: {
          control: 'boolean',
          description: 'Desativa o botão',
          defaultValue: false
        },
      }
} satisfies Meta<typeof Button>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Basic: Story = {
    args: {
      isLoading: false,
      text: 'Clique aqui'
    }
  }

export const Loading: Story = {
    args: {
        isLoading: true,
        text: 'Carregando...'
    }
}

export const Disabled: Story = {
    args: {
        isLoading: false,
        text: 'Desativado',
        disabled: true
    }
}
