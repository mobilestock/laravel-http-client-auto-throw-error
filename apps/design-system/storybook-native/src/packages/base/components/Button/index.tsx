import React, { PropsWithChildren } from 'react'
import { ActivityIndicator, StyleProp, TextStyle, TouchableHighlightProps } from 'react-native'
import styled, { css } from 'styled-components/native'

import { theme } from '../../utils/theme'

export interface PropsButton extends TouchableHighlightProps {
  text?: string
  isLoading?: boolean
  textStyle?: StyleProp<TextStyle>
}

export const Button: React.FC<PropsWithChildren<PropsButton>> = props => {
  return (
    <ButtonStyle {...props}>
      {props.isLoading ? (
        <ActivityIndicator color={theme.colors.decorator.pure} size={25} />
      ) : (
        <>{props.children ? props.children : <Text style={props.textStyle}>{props.text}</Text>}</>
      )}
    </ButtonStyle>
  )
}

const ButtonStyle = styled.TouchableHighlight`
  background-color: ${theme.colors.button.base};
  min-height: ${theme.layout.height(2)}px;
  justify-content: center;
  align-items: center;
  border-radius: ${theme.layout.size(1)}px;
  margin: ${theme.layout.size(1)}px;
  padding: ${theme.layout.size(1)}px;

  ${({ disabled }) =>
    disabled &&
    css`
      opacity: 0.5;
    `}
`

const Text = styled.Text`
  color: ${theme.colors.text.secondary};
  font-size: ${theme.fonts.size(3)}px;
  text-align: center;
`
