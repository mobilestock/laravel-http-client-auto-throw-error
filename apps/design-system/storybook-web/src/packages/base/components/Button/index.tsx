import React, { ButtonHTMLAttributes } from 'react'
import styled from 'styled-components'

import { CircularProgress } from '@mui/material'

import { theme } from '../../utils/theme'

export interface PropsButton extends ButtonHTMLAttributes<HTMLButtonElement> {
  text?: string
  isLoading?: boolean
}

export const Button: React.FC<PropsButton> = props => {
  return (
    <ButtonStyle {...props}>
      <span className="emphasis">
        {props.isLoading ? <CircularProgress className="circular" /> : <>{props.text}</>}
      </span>
      {props.isLoading || props.children}
    </ButtonStyle>
  )
}

const ButtonStyle = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;

  width: 100%;
  min-height: 3rem;
  padding: 0.9rem 1rem;

  font-size: 0.9rem;
  font-family: 'Open Sans', sans-serif;
  font-weight: 400;

  cursor: pointer;

  box-shadow: 0 0.25rem 0.25rem rgba(0, 0, 0, 0.1);
  border: none;
  border-radius: 0.4rem;
  background-color: ${theme.colors.button.base};
  color: ${theme.colors.text.secondary};
  svg {
    font-size: 1.6rem;
    margin-right: 0.3rem;
  }
  &:hover {
    opacity: 0.8;
  }
  .emphasis {
    text-transform: uppercase;
  }
  .circular {
    width: 1rem !important;
    height: 1rem !important;
    color: ${theme.colors.text.secondary};
    svg {
      margin-right: 0;
    }
  }
  &:disabled {
    opacity: 0.7;
  }
`
