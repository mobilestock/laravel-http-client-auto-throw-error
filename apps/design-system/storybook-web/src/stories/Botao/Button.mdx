import { Canvas, Meta, Story } from '@storybook/blocks';
import * as ButtonStories from './index.stories';

<Meta of={ButtonStories} />

# Botão

O componente **Botão** é utilizado para ações gerais em formulários e interfaces, suportando variações como `Carregando`, `Desativado`, e diferentes estilos.

## Uso Básico

Você pode utilizar o botão com um texto padrão e sem estado de carregamento.

<Canvas of={ButtonStories.Primary} />

## Estados do Botão

### Carregando

Use a prop `isLoading` para indicar um estado de carregamento no botão.

<Canvas of={ButtonStories.Loading} />

### Desativado

Use a prop `disabled` para desativar o botão e impedir interações.

<Canvas of={ButtonStories.Disabled} />
