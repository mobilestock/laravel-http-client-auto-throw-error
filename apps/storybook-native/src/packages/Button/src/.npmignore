*.stories.tsx
*.test.tsx
__snapshots__
