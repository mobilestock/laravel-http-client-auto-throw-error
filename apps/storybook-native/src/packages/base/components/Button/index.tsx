import React, { PropsWithChildren } from 'react';
import { ActivityIndicator, StyleProp, TextStyle, TouchableHighlightProps } from 'react-native';
import styled, { css } from 'styled-components/native';

import { theme } from '../../utils/theme';

export interface PropsButton extends TouchableHighlightProps {
  text?: string;
  isLoading?: boolean;
  textStyle?: StyleProp<TextStyle>;
}

export const Button: React.FC<PropsWithChildren<PropsButton>> = (props) => {
  return (
    <ButtonStyle {...props}>
      {props.isLoading ? (
        <ActivityIndicator color={theme.colors.container.default} size={25} />
      ) : (
        <>{props.children ? props.children : <Text style={props.textStyle}>{props.text}</Text>}</>
      )}
    </ButtonStyle>
  );
};

const ButtonStyle = styled.TouchableHighlight`
  background-color: ${({ theme }) => theme.colors.button.default};
  min-height: ${({ theme }) => theme.layout.height(2)}px;
  justify-content: center;
  align-items: center;
  border-radius: ${({ theme }) => theme.layout.size(2)}px;
  margin: ${({ theme }) => theme.layout.size(1)}px;
  padding: ${({ theme }) => theme.layout.size(2)}px;

  ${({ disabled }) =>
    disabled &&
    css`
      opacity: 0.5;
    `}
`;

const Text = styled.Text`
  color: ${({ theme }) => theme.colors.text.secondary};
  font-size: ${({ theme }) => theme.fonts.size(16)}px;
  text-align: center;
`;
