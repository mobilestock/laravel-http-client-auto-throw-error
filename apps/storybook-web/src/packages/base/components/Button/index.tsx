import { CircularProgress } from '@mui/material'
import React, { ButtonHTMLAttributes } from 'react'
import styled from 'styled-components'

export interface PropsButton extends ButtonHTMLAttributes<HTMLButtonElement> {
  text?: string
  isLoading?: boolean
}

export const Button: React.FC<PropsButton> = (props) => {
  return (
    <ButtonStyle {...props}>
      <span className="emphasis">
        {props.isLoading ? <CircularProgress className="circular" /> : <>{props.text}</>}
      </span>
      {props.isLoading || props.children}
    </ButtonStyle>
  )
}

const ButtonStyle = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;

  width: 100%;
  min-height: 3rem;
  padding: 0.9rem 1rem;

  font-size: 0.9rem;
  font-family: 'Open Sans', sans-serif;
  font-weight: 400;

  cursor: pointer;

  box-shadow: 0 0.25rem 0.25rem ${({ theme }) => theme.colors.button.shadow};
  border: none;
  border-radius: 0.4rem;
  background-color: ${({ theme }) => theme.colors.button.default};
  color: ${({ theme }) => theme.colors.text.default};
  svg {
    font-size: 1.6rem;
    margin-right: 0.3rem;
  }
  &:hover {
    opacity: 0.8;
  }
  .emphasis {
    text-transform: uppercase;
  }
  .circular {
    width: 1rem !important;
    height: 1rem !important;
    color: ${({ theme }) => theme.colors.text.default};
    svg {
      margin-right: 0;
    }
  }
  &:disabled {
    opacity: 0.7;
  }
`
