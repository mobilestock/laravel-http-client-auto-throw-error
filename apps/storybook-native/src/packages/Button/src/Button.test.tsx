import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import { Text } from 'react-native'
import Button from '.'

describe('Button Component - Native', () => {
  it('deve renderizar sem erros', () => {
    render(global.app(<Button />))
  })

  it('deve exibir o texto passado via props', () => {
    const { getByText } = render(global.app(<Button text="Clique aqui" />))
    expect(getByText('Clique aqui')).toBeTruthy()
  })

  it('deve renderizar os filhos quando passados', () => {
    const { getByText } = render(global.app(
      <Button>
        <Text>Enviar</Text>
      </Button>,
    ))
    expect(getByText('Enviar')).toBeTruthy()
  })

  it('deve mostrar o indicador de carregamento quando isLoading for true', () => {
    const { getByTestId } = render(global.app(<Button isLoading={true} />))
    expect(getByTestId('loading-indicator')).toBeTruthy()
  })

  it('nÃ£o deve mostrar o indicador de carregamento quando isLoading for false', () => {
    const { queryByTestId } = render(global.app(<Button isLoading={false} />))
    expect(queryByTestId('loading-indicator')).toBeNull()
  })

  it('deve chamar onPress quando pressionado', () => {
    const onPressMock = jest.fn()
    const { getByTestId } = render(global.app(<Button onPress={onPressMock} />))
    fireEvent.press(getByTestId('button'))
    expect(onPressMock).toHaveBeenCalled()
  })

  it('deve estar desabilitado quando a prop disabled for true', () => {
    const { getByTestId } = render(global.app(<Button disabled={true} />))
    const button = getByTestId('button')

    expect(button.props.accessibilityState.disabled).toBe(true)
  })

  it('deve aplicar o estilo personalizado quando textStyle for fornecido', () => {
    const customStyle = { fontSize: 20 }
    const { getByText } = render(global.app(<Button text="Estilo Personalizado" textStyle={customStyle} />))
    expect(getByText('Estilo Personalizado').props.style).toEqual(
      expect.arrayContaining([expect.objectContaining(customStyle)]),
    )
  })
})
