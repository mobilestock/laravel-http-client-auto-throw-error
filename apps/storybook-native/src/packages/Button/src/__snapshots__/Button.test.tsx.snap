// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component - Native deve corresponder ao snapshot 1`] = `
<View
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#029afa",
      "borderBottomLeftRadius": 3.6,
      "borderBottomRightRadius": 3.6,
      "borderTopLeftRadius": 3.6,
      "borderTopRightRadius": 3.6,
      "justifyContent": "center",
      "marginBottom": 1.8,
      "marginLeft": 1.8,
      "marginRight": 1.8,
      "marginTop": 1.8,
      "minHeight": 26.5,
      "paddingBottom": 3.6,
      "paddingLeft": 3.6,
      "paddingRight": 3.6,
      "paddingTop": 3.6,
    }
  }
  testID="button"
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 27,
        "textAlign": "center",
      }
    }
  />
</View>
`;
